dashboard/app/controllers/certificate_images_controller.rb

```diff
@@ -0,0 +1,27 @@
+require 'base64'
+
+class CertificateImagesController < ApplicationController
+  # GET /certificate_images/filename.jpg
+  def show
+    filename = params[:filename]
+    format = params[:format]
+    unless ['jpg', 'jpeg', 'png'].include?(format)
+      return render status: :bad_request, json: {message: "invalid format: #{format}"}
+    end
+
+    begin
+      data = JSON.parse(Base64.urlsafe_decode64(filename))
+    rescue ArgumentError, OpenSSL::Cipher::CipherError
+      return render status: :bad_request, json: {message: 'invalid base64'}
+    end
+
+    begin
+      image = create_course_certificate_image(data['name'], data['course'], data['sponsor'], data['course_title'])
+      image.format = format
+      content_type = "image/#{format}"
+      send_data image.to_blob, type: content_type
+    ensure
+      image && image.destroy!
+    end
+  end
+end
```

Comment:
much of this api logic was copied from the existing pegasus APIs such as these two:
* https://github.com/code-dot-org/code-dot-org/blob/1bcdf1f1258829cba969546c0875a69c42d71dcf/pegasus/routes/hoc_routes.rb#L110
* https://github.com/code-dot-org/code-dot-org/blob/1bcdf1f1258829cba969546c0875a69c42d71dcf/pegasus/routes/hoc_routes.rb#L84

fortunately, most of the work to generate the images is done in `create_course_certificate_image` which was already accessible from dashboard: https://github.com/code-dot-org/code-dot-org/blob/9184f8aa020d75bbd63311b5a1305dff6d84bafd/lib/cdo/graphics/certificate_image.rb#L143